ticketutil/jira.py

```diff
@@ -71,25 +72,50 @@ def _verify_project(self, project):
                 logger.error(e)
             return False
 
-    def _verify_ticket_id(self, ticket_id):
+    def get_ticket_content(self, ticket_id=None):
         """
-        Queries the JIRA API to see if ticket_id is a valid ticket for the given JIRA instance.
-        :param ticket_id: The ticket you're verifying.
-        :return: True or False depending on if ticket is valid.
+        Queries the JIRA API to get ticket_content using ticket_id.
+
+        :param ticket_id: ticket number, if not set self.ticket_id is used.
+        :return: self.request_result: Named tuple containing request status, error_message, and url info.
         """
+        if ticket_id is None:
+            ticket_id = self.ticket_id
+            if not self.ticket_id:
+                error_message = "No ticket ID associated with ticket object. " \
+                                "Set ticket ID with set_ticket_id(<ticket_id>)"
+                logger.error(error_message)
+                return self.request_result._replace(status='Failure', error_message=error_message)
+
         try:
             r = self.s.get("{0}/{1}".format(self.rest_url, ticket_id))
-            logger.debug("Verify ticket_id: status code: {0}".format(r.status_code))
+            logger.debug("Get ticket content: status code: {0}".format(r.status_code))
             r.raise_for_status()
-            logger.debug("Ticket {0} is valid".format(ticket_id))
-            return True
+            self.ticket_content = r.json()
+            return self.request_result._replace(ticket_content=self.ticket_content)
         except requests.RequestException as e:
             if r.json()['errorMessages'][0] == "Issue Does Not Exist":
```

Comment:
Instead of "Issue Does Not Exist", I see the following error message in r.json()['errorMessages'][0] when running get_ticket_content() on an invalid ticket_id:

`"Issue does not exist or you do not have permission to see it."`

I'm testing on my own personal JIRA dev instance and not the Red Hat Internal instance, so maybe JIRA has changed their message in a newer version? You could use the following to check for both cases:

`if "issue does not exist" in r.json()['errorMessages'][0].lower()`

Reply:
The previous works for both stage and production instances. I am not familiar with other. I changed it how you suggested.